source/Meadow.Core/Update/UpdateService.cs

```diff
@@ -651,4 +662,93 @@ public void ApplyUpdate(UpdateInfo updateInfo)
 
         State = UpdateState.Idle;
     }
+
+    private readonly ConcurrentDictionary<string, (Type CommandType, Action<object> Action)> CommandSubscriptions = new();
+    private const string UntypedCommandTypeName = "<<<MEADOWCOMMAND>>>";
+
+    void ICommandService.Subscribe(Action<MeadowCommand> action)
+    {
+        CommandSubscriptions[UntypedCommandTypeName] = (CommandType: typeof(MeadowCommand), Action: x => action((MeadowCommand)x));
+    }
+
+    void ICommandService.Subscribe<T>(Action<T> action)
```

Comment:
Subscribing to the same type multiple times, or subscribing using different types of the same name (because they are in different namespaces), will result in a last-in-wins result.

Reply:
I was just about to ask... =)